rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/RascalLanguageServer.java

```diff
@@ -140,7 +161,84 @@ public WorkspaceService getWorkspaceService() {
+         * Rascal REPLs get access to @see IDEServices to provide browsers 
+         * for interactive visualizations, starting editors and resolving IDE project URI.
+         * 
+         * @return the port number that the IDE services are running on
+         * @throws IOException when a new server socket can not be established.
+         */
+        private IDEServicesConfiguration startIDEServices(IRascalLanguageClient client) {
+            try {
+                ServerSocket socket = new ServerSocket(0);
+            
+                new IDEServicesThread(client, socket).start();
+            
+                return new IDEServicesConfiguration(socket.getLocalPort());
+            } catch (IOException e) {
+                logger.error(e);
+                throw new RuntimeException(e);
+            }
+        }
+
+        private static class IDEServicesThread extends Thread {
+            private final IRascalLanguageClient ideClient;
+            private final ServerSocket serverSocket;
+
+            public IDEServicesThread(IRascalLanguageClient client, ServerSocket socket) {
+                super("Terminal IDE Services Thread");
+                setDaemon(true);
+                this.serverSocket = socket;
+                this.ideClient = client;
+            }
+
+            @Override
+            public void run() {
+                try {
+                    while(true) {
+                        logger.trace("Accepting connection for TerminalIDE Services on port {}", serverSocket.getLocalPort());
+                        Socket connection = serverSocket.accept();
+
+                        try {
+                            Launcher<ITerminalIDEServer> ideServicesServerLauncher = new Launcher.Builder<ITerminalIDEServer>()
+                                .setLocalService(new TerminalIDEServer(ideClient))
+                                .setRemoteInterface(ITerminalIDEServer.class) // TODO this should be an empty interface?
+                                .setInput(connection.getInputStream())
+                                .setOutput(connection.getOutputStream())
+                                .setExceptionHandler(e -> {
+                                    logger.error(e);
+                                    return new ResponseError(ResponseErrorCode.InternalError, e.getMessage(), e);
+                                })
+                                .create();
+
+                            logger.trace("Terminal IDE services starts listening");
+                            ideServicesServerLauncher.startListening();
+                        }
+                        catch (Throwable e) {
+                            logger.error("Making a connection for Terminal IDE services failed", e);
+                        }
+                    }
+                }
+                catch (IOException e) {
+                    logger.error("Cannot accept new Terminal IDE Services connections", e);
```

Comment:
I think that the `.accept` call has to be moved inside the try. We don't want one broken connection setup to break the repl IDE services that come afterwards. (since this exception is rethrown and ends up terminating the thread).

Reply:
nice